.github/workflows/project_ci.yml

```diff
@@ -0,0 +1,92 @@
+# In YAML, 'raw' and 'endraw' can't come at the beginning of a token or at the end of a quoted value, which is why they're in weird places. They are
+# needed to escape YAML templating variables, which still allowing cookiecutter to replace the value of the repository name.
+
+name: Project CI
+on: pull_request
+
+jobs:
+  create_and_test_project:
+    name: Create and test Django project
+    runs-on: ubuntu-latest
+    env:
+      DATABASE_URL: "psql://postgres:postgres@localhost/postgres"
+      DJANGO_SETTINGS_MODULE: "sampleapp.config.settings.test"
+      DJANGO_SECRET_KEY: "!!!! Change me !!!!"
+      ACUITY_USER_KEY: ""
+      ACUITY_SECRET_KEY: ""
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Install and configure Poetry
+        uses: snok/install-poetry@e3dbfd357f4751d4f582d62bc8a71e56c2a7015b # v1.3.2
+        with:
+          version: latest
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+          installer-parallel: true
+
+      # Configure Python for the template
+      - uses: actions/setup-python@v4
+        id: setup_python
+        with:
+          python-version-file: ".python-version"
+          cache: "poetry"
+
+      - run: poetry env use ${{ steps.setup_python.outputs.python-version }}
+
+      - name: Load cached venv if cache exists
+        id: cached-poetry-dependencies
+        uses: actions/cache@v2
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
+
+      - name: Install dependencies if cache does not exist
+        run: poetry install --no-interaction --no-root
+
+      # Instantiate the project
+      - name: Instantiate project using the values in cookiecutter.json
+        run: poetry run cookiecutter . --no-input
+
+      # Configure the project environment
+      - uses: actions/setup-node@v3
+        with:
+          node-version: lts/*
+          cache: "npm"
+          cache-dependency-path: "**/package-lock.json"
+
+      - run: npm install
+        working-directory: ./sampleapp
+
+      - run: npm run build
+        working-directory: ./sampleapp
+
+      - run: poetry env use 3.10.4
+        working-directory: ./sampleapp
+
+      - name: Install dependencies if cache does not exist
```

Comment:
Yes and no. I am purposely  doing `poetry run cookiecutter . --no-input` so that the project is instantiated each time using the same scripts that would be run by an actual user. But the dependencies for the project are installed in a way that is cached, and that is the one slow part anyway.

Reply:
Right now the reason why the script isn't installing the dependencies on CI is just that it is conveniently erroring out in a convenient way that happens to make the whole thing work. We could formalize this by using an env variable to put the statements that error out into an `if` statement.